pkg/controller/ecr/repositorypolicy/controller.go

```diff
@@ -0,0 +1,214 @@
+	errUpdate       = "failed to update repository policy"
+	errDelete       = "failed to delete the repository resource"
+	errSpecUpdate   = "cannot update spec of repository policy custom resource"
+	errStatusUpdate = "cannot update status of repository policy custom resource"
+	errNotSpecified = "failed to format Repository Policy, no rawPolicy or policy specified"
+)
+
+// SetupRepositoryPolicy adds a controller that reconciles ECR.
+func SetupRepositoryPolicy(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter) error {
+	name := managed.ControllerName(v1alpha1.RepositoryPolicyGroupKind)
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewDefaultManagedRateLimiter(rl),
+		}).
+		For(&v1alpha1.RepositoryPolicy{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.RepositoryPolicyGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient()}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	kube client.Client
+}
+
+func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1alpha1.RepositoryPolicy)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+	cfg, err := awsclient.GetConfig(ctx, c.kube, mg, cr.Spec.ForProvider.Region)
+	if err != nil {
+		return nil, err
+	}
+	return &external{client: awsecr.New(*cfg), kube: c.kube}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client ecr.RepositoryPolicyClient
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.RepositoryPolicy)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	response, err := e.client.GetRepositoryPolicyRequest(&awsecr.GetRepositoryPolicyInput{
+		RegistryId:     cr.Spec.ForProvider.RegistryID,
+		RepositoryName: cr.Spec.ForProvider.RepositoryName,
+	}).Send(ctx)
+
+	if err != nil {
+		if ecr.IsRepoNotFoundErr(err) {
+			return managed.ExternalObservation{}, nil
+		}
+		return managed.ExternalObservation{}, awsclient.Wrap(resource.Ignore(ecr.IsPolicyNotFoundErr, err), errGet)
+	}
+
+	policyData, err := e.formatRepositoryPolicy(cr)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGet)
+	}
+
+	if response.PolicyText != nil {
+		cr.Status.AtProvider.PolicyText = *response.PolicyText
+	}
+
+	current := cr.Spec.ForProvider.DeepCopy()
+	ecr.LateInitializeRepositoryPolicy(&cr.Spec.ForProvider, response)
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, awsclient.Wrap(err, errSpecUpdate)
+		}
+	}
```

Comment:
```suggestion
```

You can return `ResourceLateInitialized: !cmp.Equal(current, &cr.Spec.ForProvider)` now with the latest crossplane-runtime.